File: src/main/java/com/caocao/baseOP/SetOP.java
package com.caocao.baseOP;

import java.util.HashSet;
import java.util.Set;

/**
 * Created by caocao on 16/10/31.
 *  1.它不允许出现重复元素；
 2.不保证集合中元素的顺序
 3.允许包含值为null的元素，但最多只能有一个null元素。
 */
public class SetOP {
    public static void main(String[] args) {
        Set a = new HashSet<String>();
        a.add("aa");
        System.out.println(a.toString());

    }
}

File: src/main/scala/com/caocao/baseOP/SetOP.scala
package com.caocao.baseOP

import scala.collection.mutable.Set;
;

/**
  * Created by m on 16-11-2.
  */
object SSetOP {
  def main(args: Array[String]): Unit = {
    println("hello = " + args)

    var s = Set[String]();
    s.add("caocao")
    s += "qinqin"
    s += "mm"
    s += "qq"
    for (singleS <- s) {
      println("singleS = " + singleS)
    }

    s -= "qinqin"
    println(s)

    println(s.exists(s => s.length == 6))

    println(s.filter(_.startsWith("c")))

    println(s.map(singleS => singleS + "").toList.sortWith((s1, s2) => s1 > s2))

    println(s.toList.sorted(Ordering.String.reverse))
  }

}
